pkgs/top-level/perl-packages.nix

```diff
@@ -914,12 +914,12 @@ let
     };
   };
 
-  AppSqitch = buildPerlModule {
-    version = "1.1.0";
+  AppSqitch = buildPerlModule rec {
+    version = "1.3.1";
     pname = "App-Sqitch";
     src = fetchurl {
-      url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.1.0.tar.gz";
```

Comment:
Not a huge fan of these as https://github.com/stigtsp/nix-cpan would override them anyway when it's finished (afaict) :/
Not that the change is bad, it's just unnecessary

Reply:
You're not a huge fan of recycling the `pname` and `version`? In which contexts would `nix-cpan` override these version strings? If I build `sqitch` in `nixpkgs` then it uses the version specified inline, here.

Reply:
It will **probably** override it when automatically updating the package next time. But @stigtsp can probably tell better than me.

Reply:
Oh, I see. Is this tool used to automatically upgrade the versions of packages used in `nixpkgs`? If so, it'd be awesome to have this tool swap out the author of `Template-Toolkit`, since `ATOOMIC`  (versus `ABW`) points to an older (apparently unmaintained) version.

Reply:
One thing to consider is that CPAN URLs doesn't always follow a `{Distribution}-v{Version}.tar.gz` file pattern.

In some cases the `v` is omitted, and other extensions like `.zip`, `.tgz`, `tar.bz2`, are sometimes used.

Another example I've found is the distribution `XML-Grove` version `4.6` which has the file name `KMACLEOD/XML-Grove-0.46alpha.tar.gz`


The correct URL and checksum can be retrieved using the metacpan api:

```
$ curl https://fastapi.metacpan.org/v1/release/XML-Grove
```

Reply:
> Is this tool used to automatically upgrade the versions of packages used in `nixpkgs`? 

Not quite yet - but you can run it against your nixpkgs checkout to update distributions. When a distribution like `Template-Toolkit` is updated, if will read the correct URL and checksums from the metacpan API, including updating the author part of the URL.

Reply:
> One thing to consider is that CPAN URLs doesn't always follow a {Distribution}-v{Version}.tar.gz file pattern.
> In some cases the v is omitted, and other extensions like .zip, .tgz, tar.bz2, are sometimes used.
> Another example I've found is the distribution XML-Grove version 4.6 which has the file name KMACLEOD/XML-Grove-0.46alpha.tar.gz

Yeah, that's kind of annoying. I took care to consider this when re-writing these URLs and made sure all packages would build.

> The correct URL and checksum can be retrieved using the metacpan api:

Oh, that's a cool tool. I checked against both of these changes and we look good:
```bash
$ for pkg in "App-Sqitch" "Template-Toolkit"; do curl -s https://fastapi.metacpan.org/v1/release/$pkg | jq .download_url; done
"https://cpan.metacpan.org/authors/id/D/DW/DWHEELER/App-Sqitch-v1.3.1.tar.gz"
"https://cpan.metacpan.org/authors/id/A/AB/ABW/Template-Toolkit-3.101.tar.gz"
```